codec-http2/src/main/java/io/netty/handler/codec/http2/HpackHuffmanDecoder.java

```diff
@@ -33,27 +33,16 @@
 
 import io.netty.buffer.ByteBuf;
 import io.netty.util.AsciiString;
-import io.netty.util.ByteProcessor;
-import io.netty.util.internal.ObjectUtil;
 import io.netty.util.internal.ThrowableUtil;
 
 import static io.netty.handler.codec.http2.Http2Error.COMPRESSION_ERROR;
 
 final class HpackHuffmanDecoder {
-
-    private static final Http2Exception EOS_DECODED = ThrowableUtil.unknownStackTrace(
-            Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - EOS Decoded",
-                    Http2Exception.ShutdownHint.HARD_SHUTDOWN), HpackHuffmanDecoder.class, "decode(..)");
-    private static final Http2Exception INVALID_PADDING = ThrowableUtil.unknownStackTrace(
-            Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - Invalid Padding",
+    private static final Http2Exception BAD_ENCODING = ThrowableUtil.unknownStackTrace(
+            Http2Exception.newStatic(COMPRESSION_ERROR, "HPACK - Bad Encoding",
                     Http2Exception.ShutdownHint.HARD_SHUTDOWN), HpackHuffmanDecoder.class, "decode(..)");
 
-    private static final Node ROOT = buildTree(HpackUtil.HUFFMAN_CODES, HpackUtil.HUFFMAN_CODE_LENGTHS);
-
-    private final DecoderProcessor processor;
-
     HpackHuffmanDecoder(int initialCapacity) {
-        processor = new DecoderProcessor(initialCapacity);
```

Comment:
Also this can now be a no-arg constructor as `initialCapacity` is not used at all.

Reply:
Was going to leave for a followup.  Transitively removing the methods that call this results in a break HTTP/2 change, ending at DefaultHttp2HeadersDecoder.   

If you are okay breaking that as part of this PR I can do it.

Reply:
I think we can break everything as part of this pr that is not public. That said you can also leave it as it is... your call.

Reply:
I removed all refs, except for those on `DefaultHttp2HeadersDecoder` and `AbstractHttp2ConnectionHandlerBuilder`.  Both of those contain no-op params now.